From d4af65336a52c6d72fdd157cf20b5e2d46d1f216 Mon Sep 17 00:00:00 2001
From: Allan Jude <allan@klarasystems.com>
Date: Tue, 21 Dec 2021 16:32:34 -0500
Subject: [PATCH 4/7] fault/auto_spare tests: zpool sync before zinject

Ensure the data has been flushed to the disk, else we read it back
from the anonymous ARC and never encounter the zinject'd faults

Signed-off-by: Allan Jude <allan@klarasystems.com>
Sponsored-by: Zededa Inc.
---
 tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh | 1 +
 tests/zfs-tests/tests/functional/fault/auto_spare_002_pos.ksh | 1 +
 2 files changed, 2 insertions(+)

diff --git a/tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh b/tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh
index a93267185..d13ace31f 100755
--- a/tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh
+++ b/tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh
@@ -82,6 +82,7 @@ for type in "mirror" "raidz" "raidz2" "draid:1s"; do
 
 	# 3. Write a file to the pool to be read back
 	log_must dd if=/dev/urandom of=$TESTFILE bs=1M count=64
+	sync_pool $TESTPOOL
 
 	# 4. Inject IO ERRORS on read with a zinject error handler
 	log_must zinject -d $FAULT -e io -T read $TESTPOOL
diff --git a/tests/zfs-tests/tests/functional/fault/auto_spare_002_pos.ksh b/tests/zfs-tests/tests/functional/fault/auto_spare_002_pos.ksh
index e9517bad7..8555f730c 100755
--- a/tests/zfs-tests/tests/functional/fault/auto_spare_002_pos.ksh
+++ b/tests/zfs-tests/tests/functional/fault/auto_spare_002_pos.ksh
@@ -70,6 +70,7 @@ for type in "mirror" "raidz" "raidz2"; do
 
 	# 3. Write a file to the pool to be read back
 	log_must dd if=/dev/urandom of=$TESTFILE bs=1M count=64
+	sync_pool $TESTPOOL
 
 	# 4. Inject CHECKSUM ERRORS on read with a zinject error handler
 	log_must zinject -d $FAULT_FILE -e corrupt -f 50 -T read $TESTPOOL
-- 
2.25.1

